### PR TITLE
feat(forward_index): run forward index in tskv

### PR DESCRIPTION
**Pull request by vrg000**

### CODE REVIEW

File: tskv/src/record_file/reader.rs [hunk 1]
```diff
@@ -2,12 +2,14 @@ use std::{borrow::Borrow, fs, hash::Hasher, io::Read, path::PathBuf};
...
 use num_traits::ToPrimitive;
+use parking_lot::Mutex;
+use direct_io::File;
 
 use super::*;
 
 pub struct Reader {
     path: PathBuf,
-    file: direct_io::File,
+    file: Mutex<File>,
```
Comment (roseboy-liu) [inline tskv/src/record_file/reader.rs:12]: dont try to lock at basic file.

**Review by roseboy-liu — COMMENTED** [review]
run "make fmt" before pull request

File: tskv/src/record_file/reader.rs (hunk 1 above)
Comment (vrg000) [reply tskv/src/record_file/reader.rs:12]: done

**Review by zipper-meng — APPROVED** [review]
:)

**Pull request merged**